app/src/main/java/com/example/universitieslist/data/remote/repository/UniversityRepositoryImpl.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.universitieslist.data.remote.repository
+
+import com.example.universitieslist.data.model.UniversityResponse
+import com.example.universitieslist.data.remote.ApiService
+import com.example.universitieslist.util.Constants.UNABLE_TO_FETCHING
+import com.example.universitieslist.util.Constants.UNKNOWN_ERROR
+import com.example.universitieslist.util.FetchUniversitiesException
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import timber.log.Timber
+import javax.inject.Inject
+
+class UniversityRepositoryImpl @Inject constructor(
+    private val apiService: ApiService
+) : UniversityRepository {
+
+    override suspend fun getUniversities(country: String): Flow<List<UniversityResponse>> = flow {
+        val unableToFetchError = "$UNABLE_TO_FETCHING $country"
+        try {
+            val response = apiService.getUniversities(country)
+            if (response.isSuccessful && response.body() != null) {
```

Comment:
Nice! But you could consider extracting this into a wrapper class or function to avoid duplicating in another network call.

Reply:
okay,